src/effects/effectknobparameterslot.cpp

```diff
@@ -239,3 +239,10 @@ void EffectKnobParameterSlot::syncSofttakeover() {
 double EffectKnobParameterSlot::getValueParameter() const {
     return m_pControlValue->getParameter();
 }
+
+double EffectKnobParameterSlot::neutralPointOnScale() const {
+    if (m_pManifestParameter) {
+        return m_pManifestParameter->neutralPointOnScale();
+    }
+    return 0.0;
```

Comment:
this looks like a std::optional candidate. wdyt?

Reply:
In this case 0.0 is a good default and valid value for the unlike case that the is no parameter connected.

Reply:
but 0.0 could also be a valid neutral point (for example in a high-pass filter), it's impossible for the caller to tell. I think the caller needs to explicitly handle the circumstance (for example by showing "error" as a value, not just 0).

Reply:
An error value for the neutral position, when we have also no value and no scale? 
This case should never happen, since the parameter knobs are hidden in that case. So coding around that is not reasonable. 

We have that state for the meta knob when no effect is loaded. Currently a 0 .. 1 knob with default is assumed and no error state. Since the value of the knob remains where is was when loading "---".  What kind of error state you have in mind?

Reply:
> An error value for the neutral position, when we have also no value and no scale?

> What kind of error state you have in mind?

The error state is if `m_pManifestParameter == nullptr`. 

> We have that state for the meta knob when no effect is loaded. Currently a 0 .. 1 knob with default 0 is assumed and no special error state. The value of the knob remains where is was when loading "---".

That doesn't seem like particularly good design IMO because that assumes some other part of the code does the appropriate checking. This assumption could get violated easily in a future refactoring. If you just look at this public member function (thus part of the class's API), there is no way to verify whether the value is returned represents an error or real value. And even if there is, the caller would have to know about that implementation detail. The result is an API that is easy to misuse.

Reply:
> Are you satisfied with 1. ?

Yes. 

>  In this particular case it does not makes sense to consider a case that never happens.
We can probably ASSERT that, I am just afraid that will bite us later.

playing devils advocate here, this sounds nonsensical. If it never happens, then assert and encode that in the type-system (plain double, no special values indicating an error state). If it happens, lets encode it in the type system (`std::optional<double>`).